python/cudf/cudf/tests/test_joining.py

```diff
@@ -1529,7 +1529,7 @@ def test_categorical_typecast_inner_one_cat(dtype):
     data = np.array([1, 2, 3], dtype=dtype)
 
     left = make_categorical_dataframe(data)
-    right = left.astype(left["key"].dtype.categories)
+    right = left.astype(left["key"].dtype.categories.dtype)
```

Comment:
These were passing by coincidence due to https://github.com/rapidsai/cudf/issues/8895